src/Mvc/Mvc/src/MvcServiceCollectionExtensions.cs

```diff
@@ -30,69 +31,156 @@ public static IMvcBuilder AddMvc(this IServiceCollection services)
+            // This method excludes all of the view-related services by default.
+            var builder = AddControllers(services);
+            if (configure != null)
             {
-                partManager.ApplicationParts.Add(new FrameworkAssemblyPart(mvcRazorAssembly));
+                builder.AddMvcOptions(configure);
             }
+
+            return builder;
         }
 
         /// <summary>
-        /// Adds MVC services to the specified <see cref="IServiceCollection" />.
+        /// Adds services for pages and controllers to the specified <see cref="IServiceCollection"/>.
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
-        /// <param name="setupAction">An <see cref="Action{MvcOptions}"/> to configure the provided <see cref="MvcOptions"/>.</param>
         /// <returns>An <see cref="IMvcBuilder"/> that can be used to further configure the MVC services.</returns>
-        public static IMvcBuilder AddMvc(this IServiceCollection services, Action<MvcOptions> setupAction)
+        public static IMvcBuilder AddRazorPages(this IServiceCollection services)
         {
             if (services == null)
             {
                 throw new ArgumentNullException(nameof(services));
             }
 
-            if (setupAction == null)
+            // This method includes what controllers need. It's not really feasible to exclude the services
+            // for controllers.
+            return services.AddControllers().AddRazorPages();
+        }
+
+        /// <summary>
+        /// Adds services for pages and controllers to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection" /> to add services to.</param>
+        /// <param name="configure">An <see cref="Action{MvcOptions}"/> to configure the provided <see cref="MvcOptions"/>.</param>
+        /// <returns>An <see cref="IMvcBuilder"/> that can be used to further configure the MVC services.</returns>
+        public static IMvcBuilder AddRazorPages(this IServiceCollection services, Action<RazorPagesOptions> configure)
+        {
+            if (services == null)
             {
-                throw new ArgumentNullException(nameof(setupAction));
+                throw new ArgumentNullException(nameof(services));
             }
 
-            var builder = services.AddMvc();
-            builder.Services.Configure(setupAction);
+            // This method includes what controllers need. It's not really feasible to exclude the services
+            // for controllers.
+            var builder = services.AddControllers().AddRazorPages();
+            if (configure != null)
+            {
+                builder.AddRazorPagesOptions(configure);
+            }
 
             return builder;
         }
 
+        internal static void AddDefaultFrameworkParts(ApplicationPartManager partManager)
```

Comment:
Unrelated, but do you know what this is for?

Reply:
Adding default framework parts?

Reply:
Yeah, but why do we add these? These aren't sources for controllers.

Reply:
I cleaned this up by renaming this method and making sure it's only included by pages/views.